docs/user_guide/source/installation/cmake.rst

```diff
@@ -1,312 +1,20 @@
-**************************************************
-Building, Testing and Installing ADIOS2 with CMake
-**************************************************
+****************************************
+Building, Testing, and Installing ADIOS2
+****************************************
 
-CMake uses "out-of-source" builds, which means keeping a separate build and source directory (different from autotools, which usually uses an in-source build).
```

Comment:
I think this is important information we need to keep somehow.

Reply:
@williamfgc : VTKm has an error message that is emitted when doing in-tree builds. How about we add that same message to the CMakeLists.txt?

Reply:
I just found this but I think there's now an idiomatic way to do it:

```
STRING(COMPARE EQUAL "${CMAKE_SOURCE_DIR}" "${CMAKE_BINARY_DIR}" insource)
IF(insource)
   MESSAGE(FATAL_ERROR "Do not build in your source dir please :)")
ENDIF(insource)
```

Reply:
@NAThompson we do that. But truly `out-of-source` is not even inside a `build` dir that lives inside the source. It's good to reinforce `CMake` good practices up front, this is supposed to be a step-by-step.